mbtiles/src/mbtiles.rs

```diff
@@ -223,7 +300,95 @@ impl Mbtiles {
+        .expect("The file size of the MBTiles file shouldn't be None") as u64;
+
+        let page_size = query!("PRAGMA page_size;")
+            .fetch_one(&mut *conn)
+            .await?
+            .page_size
+            .unwrap() as u64;
+        let tile_infos_query = query!(
+            "
+    SELECT zoom_level             AS zoom,
+           count()                AS count,
+           min(length(tile_data)) AS smallest,
+           max(length(tile_data)) AS largest,
+           avg(length(tile_data)) AS average,
+           min(tile_column)       AS min_tile_x,
+           min(tile_row)          AS min_tile_y,
+           max(tile_column)       AS max_tile_x,
+           max(tile_row)          AS max_tile_y
+    FROM tiles
+    GROUP BY zoom_level"
+        );
+        let mbt_type = self.detect_type(&mut *conn).await?;
+        let level_rows = tile_infos_query.fetch_all(&mut *conn).await?;
+        let level_details: Vec<ZoomStats> = level_rows
+            .into_iter()
+            .map(|r| {
+                let zoom = r.zoom.unwrap() as u8;
+                let count = r.count as u64;
+                let tile_length = 40075016.7 / (2_u32.pow(zoom as u32)) as f64;
+
+                let smallest = r.smallest.unwrap_or(0) as u64;
+                let largest = r.largest.unwrap_or(0) as u64;
+                let average = r.average.unwrap_or(0.0);
+
+                let min_tile_x = r.min_tile_x.unwrap();
+                let min_tile_y = r.min_tile_y.unwrap();
+                let max_tile_x = r.max_tile_x.unwrap();
+                let max_tile_y = r.max_tile_y.unwrap();
+
+                let (minx, miny) = webmercator_to_wgs84(
+                    -20037508.34 + min_tile_x as f64 * tile_length,
+                    -20037508.34 + min_tile_y as f64 * tile_length,
+                );
+                let (maxx, maxy) = webmercator_to_wgs84(
+                    -20037508.34 + (max_tile_x as f64 + 1.0) * tile_length,
+                    -20037508.34 + (max_tile_y as f64 + 1.0) * tile_length,
+                );
 
+                let bbox = Bounds::new(minx, miny, maxx, maxy);
+                ZoomStats {
+                    zoom,
+                    count,
+                    smallest,
+                    largest,
+                    average,
+                    bbox,
+                }
+            })
+            .collect();
+
+        let count = level_details.iter().map(|l| l.count).sum();
+        let smallest = level_details.iter().map(|l| l.smallest).reduce(u64::min);
+        let largest = level_details.iter().map(|l| l.largest).reduce(u64::max);
+        let average = level_details
+            .iter()
+            .map(|l| l.average * l.count as f64)
+            .sum::<f64>()
+            / count as f64;
+        Ok(Statistics {
+            file_path: self.filepath.clone(),
+            file_size,
+            schema: mbt_type,
+            page_size,
+            zoom_stats_list: level_details,
```

Comment:
you would want to rename `level_details` to be the same as the field name - this way no need for the `: level_details` part.  Same for the `schema`. Naming things same as in struct usually keeps things a bit cleaner.

Reply:
P.S. `mbt_type` is probably a bit better for the `Statistics` struct too.  `schema` is more ambiguous.

Reply:
Got it and udpated.